pipeline/mist.go

```diff
@@ -41,21 +41,21 @@ func (m *mist) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 	if !isVideo(job.RequestID, job.SourceFile) {
 		return nil, fmt.Errorf("source was not a video: %s", job.SourceFile)
 	}
-	// Arweave URLs don't support HTTP Range requests and so Mist can't natively handle them for segmenting
+	// Arweave / IPFS URLs don't support HTTP Range requests and so Mist can't natively handle them for segmenting
 	// This workaround copies the file from Arweave to S3 and then tells Mist to use the S3 URL
-	if clients.IsArweaveOrIPFSURL(job.SourceFile) {
+	if clients.IsDStorageResource(job.SourceFile) {
```

Comment:
Right, here we need to do this copying in case of a gateway URL as well. But maybe not all places that use this helper want that.